src/pcre2_compile.c

```diff
@@ -4301,10 +4297,18 @@ while (ptr < ptrend)
                 errorcode = ERR15;
                 goto FAILED;
                 }
-              *parsed_pattern++ = meta ? META_SCS_NEXT_NUMBER : META_SCS_NUMBER;
-              offset = (PCRE2_SIZE)(ptr - cb->start_pattern - 2);
-              PUTOFFSET(offset, parsed_pattern);
+
+              if (meta == 0 || (next_offset - offset) >= 0x10000)
+                {
+                *parsed_pattern++ = META_OFFSET;
+                PUTOFFSET(next_offset, parsed_pattern);
+                offset = next_offset;
+                }
+
+              *parsed_pattern++ =
+                META_SCS_NUMBER | (uint32_t)(next_offset - offset);
```

Comment:
It looks like the old code was writing out the offset of the `ptr` after `read_number`, so pointing to the comma following, whereas the new code is writing out the offset of the start of the number.

I don't especially mind, but I've been trying to make error messages point after the syntax error, in the style of Perl which says "error was left of <-- HERE".

Reply:
The old code used `(ptr - cb->start_pattern - 2)` which was copied from conditional blocks and does not make much sense. 

```
  re> /(?(1))/
Failed: error 115 at offset 2: reference to non-existent subpattern
  re> /(?(<A>))/
Failed: error 115 at offset 4: reference to non-existent subpattern
```

I though I just follow the second variant all the time. Anyway, this could be made consistent in another patch. So far nobody cared too much about error offsets, but this is probably important for you.

Reply:
To be honest, it's not _important_ to me, I'm just being fussy. If I want to change the error offsets to match Perl, I'll do it another time. Don't worry about it now!